mechanicalsoup/form.py

```diff
@@ -277,6 +277,24 @@ def set(self, name, value, force=False):
             return
         raise LinkNotFoundError("No valid element named " + name)
 
+    def __getitem__(self, name):
+        """Forwards arguments to :func:`~Form.get` to get value for a given form field
+        """
+        return self.get(name)
+
+    def get(self, name):
+        i = self.form.find("input", {"name": name})
+        if not i:
+            raise LinkNotFoundError("No input field named " + name)
```

Comment:
We've already been using `LinkNotFoundError` for too many things, but I think it's time to stop this proliferation.

Should we add a `ElementNotFoundError` or so?

Reply:
I had KeyError in mind, but then found this.
https://github.com/MechanicalSoup/MechanicalSoup/blob/b6307dfea36a57a3a8a97674e72829d3f07b890e/mechanicalsoup/utils.py#L12-L14

So I kept it for consistency. Indeed, its good idea to use ElementNotFound Exception, but again for consistency .set method should use it too?

Reply:
We started using `LinkNotFoundError` in places where we shouldn't have. The issue is backward compatibility, which forces us to keep code catching `LinkNotFoundError` working with future versions of the library.

Maybe we should just declare a new `MechanicalSoupError` alias for `LinkNotFoundError`, and use it. We'd keep consistency, backward compatibility, but we wouldn't have distinct exceptions for specific cases.

Reply:
This one is not easy to fix, it might require additional discussion. I would use MechanicalSoupError as base exception class, with derived classes like LinkNotFoundError, ElementNotFoundError and InvalidFormMethod. I would prefer to use separate PR for such change, otherwise this one might never be finished. 🙂

Reply:
I was pointing it out because the longer we wait, the harder it may be to fix later, but you're right that it shouldn't block merging this PR.